www/main/navigator/misc/components/RecordPane.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useStoreState, useStoreActions } from 'easy-peasy';
+import { ipcRenderer } from 'electron/renderer';
+import anvaad from 'anvaad-js';
+
+import { generateSRT } from 'subtitle-generator';
+
+const path = require('path');
+const remote = require('@electron/remote');
+const { i18n } = remote.require('./app');
+const fs = require('fs');
+
+export const RecordPane = ({ className }) => {
+  const { isSubtitleRecording, subtitleData } = useStoreState((state) => state.navigator);
+  const { setIsSubtitleRecording } = useStoreActions((state) => state.navigator);
+  const [fileList, setFileList] = useState([]);
+
+  const userData = remote.app.getPath('userData');
+  const directory = path.resolve(userData, 'subtitles');
+
+  const startRecording = () => {
+    if (!isSubtitleRecording) {
+      setIsSubtitleRecording(true);
+    }
+  };
+
+  const stopRecording = () => {
+    const parsed = subtitleData.map((data, index) => {
+      return {
+        id: index,
+        seconds: data.seconds,
+        content: anvaad.unicode(data.content),
+      };
+    });
+    const srt = generateSRT(parsed, 'seconds');
+    if (isSubtitleRecording) {
+      setIsSubtitleRecording(false);
+    }
+    if (!fs.existsSync(directory)) {
+      fs.mkdirSync(directory);
+    }
+    const currentDate = new Date();
+    const filename = `${currentDate.getFullYear()}-${currentDate.getMonth()}-${currentDate.getDate()}-${currentDate.getHours()}${currentDate.getMinutes()}${currentDate.getSeconds()}.srt`;
```

Comment:
veerji this string is way too lengthy. It should not cross more than 100 letters.